PWGHF/vertexingHF/AliAnalysisTaskSEXic0pp13TeV.cxx

```diff
@@ -0,0 +1,1514 @@
+#include "AliTOFPIDResponse.h"
+#include "AliAODPidHF.h"
+#include "AliInputEventHandler.h"
+#include "AliPPVsMultUtils.h"
+#include "AliESDtrackCuts.h"
+#include "AliESDtrack.h"
+#include "AliAODInputHandler.h"
+#include "AliAnalysisTaskSEHFTreeCreator.h"
+#include "AliMultSelection.h"
+
+
+const Double_t pi = TMath::Pi();
+enum {  kPN=1, kPP, kNN, kMixing, kAllType}; //P=Positive charge, N=Negative
+
+AliAnalysisTaskSEXic0RunTable::AliAnalysisTaskSEXic0RunTable() :
+  fCollisionType(kUnknownCollType){;}   //defult constructor
+
+AliAnalysisTaskSEXic0RunTable::AliAnalysisTaskSEXic0RunTable(Int_t fRunNumber){
+  //if (fRunNumber>=256504 && fRunNumber<=260014) fCollisionType=kPP; //LHC16kl
+  //else if (fRunNumber>=254604 && fRunNumber<=255467) fCollisionType=kPP; //LHC16h
+  fCollisionType=kPP;
+}
+AliAnalysisTaskSEXic0RunTable::~AliAnalysisTaskSEXic0RunTable() {;}
+
+//___________________________________________________________________
+AliAnalysisTaskSEXic0pp13TeV::AliAnalysisTaskSEXic0pp13TeV() :
+  AliAnalysisTaskSE("AliAnalysisTaskSEXic0pp13TeV"),
+  fOption(){
+    DefineOutput (1, TDirectory::Class());
+    DefineOutput (2, TList::Class());
+    DefineOutput (3, TTree::Class());
+    DefineOutput (4, TTree::Class());
+    DefineOutput (5, TTree::Class());
+    DefineOutput (6, TTree::Class());
+    DefineOutput (7, AliNormalizationCounter::Class());
+    fMCCutTreeVariable = 0;
+    fPaireXiTreeVariable = 0;
+    fMCTreeVariable = 0;
+    fEventTreeVariable = 0;
+    fMCCutTree = 0;
+    fPaireXiTree = 0;
+    fMCTree = 0;
+    fEventTree = 0;
+}
+//___________________________________________________________________
+AliAnalysisTaskSEXic0pp13TeV::AliAnalysisTaskSEXic0pp13TeV( const char *name, const char *option ) :
+  AliAnalysisTaskSE(name),
+  fOption(option){
+    DefineOutput (1, TDirectory::Class());
+    DefineOutput (2, TList::Class());
+    DefineOutput (3, TTree::Class());
+    DefineOutput (4, TTree::Class());
+    DefineOutput (5, TTree::Class());
+    DefineOutput (6, TTree::Class());
+    DefineOutput (7, AliNormalizationCounter::Class());
+    fMCCutTreeVariable = 0;
+    fPaireXiTreeVariable = 0;
+    fMCTreeVariable = 0;
+    fEventTreeVariable = 0;
+    fMCCutTree = 0;
+    fPaireXiTree = 0;
+    fMCTree = 0;
+    fEventTree = 0;
+}
+//___________________________________________________________________
+AliAnalysisTaskSEXic0pp13TeV::AliAnalysisTaskSEXic0pp13TeV( const AliAnalysisTaskSEXic0pp13TeV& ap ) :
+  fOption(ap.fOption) {;}
+
+//___________________________________________________________________
+AliAnalysisTaskSEXic0pp13TeV& AliAnalysisTaskSEXic0pp13TeV::operator = ( const AliAnalysisTaskSEXic0pp13TeV& ap ) {
+  // assignment operator
+  this->~AliAnalysisTaskSEXic0pp13TeV();
+  new(this) AliAnalysisTaskSEXic0pp13TeV(ap);
+  return *this;
+}
+//___________________________________________________________________
+AliAnalysisTaskSEXic0pp13TeV::~AliAnalysisTaskSEXic0pp13TeV(){
+  delete fOutput;
```

Comment:
The destructor is not deleting `fHistos` and all the the objects inside it, which are created with new, since you use `THistManager::CreateTH1` and `THistManager::CreateTH2` methods (see here https://github.com/alisw/AliPhysics/blob/62aec2a74e94bcdf3408f4d72f0149a9af008a73/PWG/Tools/THistManager.cxx#L80). You should add `delete fHistos` otherwise you would have meamory leaks that could make several jobs fail.